### PR TITLE
Add poem create, detail, edit, delete, like, unlike

### PR DESCRIPTION
**Pull request by HinataIzayoi**
Let User create Poem, select align and privacy, and view Poem detail.
User can edit, delete, like or unlike Poem.

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -88,14 +89,73 @@ export default class App extends React.Component {
   async userDetail({token}) {
     const user = await this.genericApi1('/v1/users/detail', {token})
         .catch((e)=>{
-          console.log(e);
+          throw new Error(e);
         });
     if (user) {
-      console.log(user);
       this.setState({user: user.payload});
+      console.log(this.state.user);
```
Comment (Kailang) [inline src/App.js]: Remove console.log.

File: src/App.js [hunk 2]
```diff
@@ -88,14 +89,73 @@ export default class App extends React.Component {
...
+    }
+  }
+
+  async userCreatePoem({title, body, visibility, token, align}) {
+    const poem = await this.genericApi1('/v1/poems/create', {title, body, visibility, align, token})
+        .catch((e)=>{
```
Comment (Kailang) [inline src/App.js]: Remove catch.

File: src/App.js [hunk 3]
```diff
@@ -88,14 +89,73 @@ export default class App extends React.Component {
...
+          throw new Error(e);
+        });
+    if (poem) {
+      console.log(poem);
```
Comment (Kailang) [inline src/App.js]: Remove console.log

File: src/App.js [hunk 4]
```diff
@@ -88,14 +89,73 @@ export default class App extends React.Component {
...
+      this.history.push(`/poems/${poem.payload}`);
     }
   }
 
+  async poemDetail({poemId, token}) {
+    const poem = await this.genericApi1('/v1/poems/detail', {poemId, token})
+        .catch((e)=>{
```
Comment (Kailang) [inline src/App.js]: Remove all .catch in this file.

File: src/components/Navbar.js [hunk 5]
```diff
@@ -61,8 +66,8 @@ export default class Navbar extends React.Component {
         <h1 class="Fz(26px) Lh(50px) D(ib) Cur(p)"><Link className="C(white) C(white):h Td(n):h" to="/">Scarletea</Link></h1>
         <span className="Fl(end) Lh(50px)">
           { isLoggedIn &&
-          <span class="C(gray) C(white):h Cur(p) Mend(30px)" onClick={this.redirectToEdit}><i class="fas fa-edit"></i></span>}
-          {/* { isLoggedIn &&  */}
+          <span class="C(gray) C(white):h Cur(p) Mend(30px)" onClick={this.redirectToWrite}><i class="fas fa-edit"></i></span>}
+          { isLoggedIn &&
```
Comment (Kailang) [inline src/components/Navbar.js]: Remove the space.

File: src/components/Navbar.js [hunk 6]
```diff
@@ -61,8 +66,8 @@ export default class Navbar extends React.Component {
...
           <span class="C(gray) C(white):h Cur(p) Mend(30px) Pos(r) D(ib)">
```
Comment (Kailang) [inline src/components/Navbar.js]: Indent this.

File: src/components/PoemDetailPage.js [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+      likeCount: 0,
+      viewCount: 0,
+      commentCount: 0,
+    };
+    this.redirectToEdit = this.redirectToEdit.bind(this);
+  }
+
+  async componentDidMount() {
+    const poem = await this.app.poemDetail({poemId: this.props.match.params.poemId, token: this.app.state.token});
+    console.log(poem);
```
Comment (Kailang) [inline src/components/PoemDetailPage.js]: Remove console log.

File: src/components/PoemDetailPage.js [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+      </div>;
+    }
+
+    const {align, title, body, visibility, likeCount, viewCount, commentCount} = this.state;
+    const writtenDateFormatted = formatDateTime(this.state.writtenDate);
+    const displayName = this.app.state.user ? this.app.state.user.displayName : '';
+
+    return <div class="My(50px) Maw(500px) Mx(a)">
+      <div>
+
```
Comment (Kailang) [inline src/components/PoemDetailPage.js]: Remove empty line.

File: src/components/PoemDetailPage.js [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+        <div class={getAlignStyle(align)}>
+          <span class="C(gray) Fz(8px)">{writtenDateFormatted}</span>
+        </div>
+        <h3 class={`Fz(24px) `+ getAlignStyle(align)}>
+          {title}
+        </h3>
+        <p class={`Whs(pw) `+ getAlignStyle(align)}>
+          {body}
+        </p>
+
```
Comment (Kailang) [inline src/components/PoemDetailPage.js]: Remove empty line.

File: src/components/PoemInfo.js [hunk 10]
```diff
@@ -12,7 +46,9 @@ export default class PoemInfo extends React.Component {
           <div class="D(ib) Va(m) Fz(12px) C(gray)">
              Written by <br/><span class="Fz(16px) C(darkred) Td(u):h">{author}</span>
           </div>
+          {!isOwner&&
```
Comment (Kailang) [inline src/components/PoemInfo.js]: {!isOwner &&

File: src/components/PoemInfo.js [hunk 11]
```diff
@@ -12,7 +46,9 @@ export default class PoemInfo extends React.Component {
...
           <span class="Fl(end) Bgc(lightgray) Bgc(darkred):h C(white):h Px(8px) Py(2px) Mt(10px) Mend(8px) Fz(12px) Bdrs($bdrs-control)">Follow</span>
```
Comment (Kailang) [inline src/components/PoemInfo.js]: Indent.

File: src/App.js (hunk 1 above)
Comment (HinataIzayoi) [reply src/App.js]: Done.

File: src/App.js (hunk 2 above)
Comment (HinataIzayoi) [reply src/App.js]: Done.

File: src/App.js (hunk 3 above)
Comment (HinataIzayoi) [reply src/App.js]: Done.

File: src/App.js (hunk 4 above)
Comment (HinataIzayoi) [reply src/App.js]: Done.

File: src/components/Navbar.js (hunk 5 above)
Comment (HinataIzayoi) [reply src/components/Navbar.js]: Done.

File: src/components/Navbar.js (hunk 6 above)
Comment (HinataIzayoi) [reply src/components/Navbar.js]: Done.

File: src/components/PoemDetailPage.js (hunk 7 above)
Comment (HinataIzayoi) [reply src/components/PoemDetailPage.js]: Done.

File: src/components/PoemDetailPage.js (hunk 8 above)
Comment (HinataIzayoi) [reply src/components/PoemDetailPage.js]: Done.

File: src/components/PoemDetailPage.js (hunk 9 above)
Comment (HinataIzayoi) [reply src/components/PoemDetailPage.js]: Done.

File: src/components/PoemInfo.js (hunk 10 above)
Comment (HinataIzayoi) [reply src/components/PoemInfo.js]: Done.

File: src/components/PoemInfo.js (hunk 11 above)
Comment (HinataIzayoi) [reply src/components/PoemInfo.js]: Done.

**Pull request merged**